Hi,

It turns out the -fpermissive was related to the thing in rwx.cpp. Originally when I tried to compile I got these errors:

<pre>rwx.cpp: In function ‘void rb_define_attr_method_base(VALUE, const string&, VALUE (*)(...), VALUE (*)(...), bool)’:
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:61: error: ‘missing’ is not a constant expression
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                             ^
rwx.cpp:229:3: note: in expansion of macro ‘rb_define_method’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |   ^~~~~~~~~~~~~~~~
rwx.cpp:229:70: note: in template argument for type ‘int’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |                                                              ~~~~~~~~^~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:60: note: in definition of macro ‘rb_define_method’
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                            ^
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: error: expression list treated as compound expression in initializer [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
rwx.cpp:229:3: note: in expansion of macro ‘rb_define_method’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |   ^~~~~~~~~~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: error: invalid conversion from ‘VALUE (*)(...)’ {aka ‘long unsigned int (*)(...)’} to ‘int’ [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
      |                                                                              |
      |                                                                              VALUE (*)(...) {aka long unsigned int (*)(...)}
rwx.cpp:229:3: note: in expansion of macro ‘rb_define_method’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |   ^~~~~~~~~~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:61: error: ‘missing’ is not a constant expression
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                             ^
rwx.cpp:231:3: note: in expansion of macro ‘rb_define_method’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |   ^~~~~~~~~~~~~~~~
rwx.cpp:231:78: note: in template argument for type ‘int’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |                                                                      ~~~~~~~~^~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:60: note: in definition of macro ‘rb_define_method’
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                            ^
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: error: expression list treated as compound expression in initializer [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
rwx.cpp:231:3: note: in expansion of macro ‘rb_define_method’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |   ^~~~~~~~~~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: error: invalid conversion from ‘VALUE (*)(...)’ {aka ‘long unsigned int (*)(...)’} to ‘int’ [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
      |                                                                              |
      |                                                                              VALUE (*)(...) {aka long unsigned int (*)(...)}
rwx.cpp:231:3: note: in expansion of macro ‘rb_define_method’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |   ^~~~~~~~~~~~~~~~
make: *** [Makefile:237 : rwx.o] Erreur 1
</pre>

So I added -fpermissive since it was suggested by the compiler's output, but it didn't clear all the errors - only some of them were transformed into warnings, and I still had that:

<pre>rwx.cpp: In function ‘void rb_define_attr_method_base(VALUE, const string&, VALUE (*)(...), VALUE (*)(...), bool)’:
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:61: error: ‘missing’ is not a constant expression
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                             ^
rwx.cpp:229:3: note: in expansion of macro ‘rb_define_method’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |   ^~~~~~~~~~~~~~~~
rwx.cpp:229:70: note: in template argument for type ‘int’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |                                                              ~~~~~~~~^~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:60: note: in definition of macro ‘rb_define_method’
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                            ^
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: warning: expression list treated as compound expression in initializer [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
rwx.cpp:229:3: note: in expansion of macro ‘rb_define_method’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |   ^~~~~~~~~~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: warning: invalid conversion from ‘VALUE (*)(...)’ {aka ‘long unsigned int (*)(...)’} to ‘int’ [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
      |                                                                              |
      |                                                                              VALUE (*)(...) {aka long unsigned int (*)(...)}
rwx.cpp:229:3: note: in expansion of macro ‘rb_define_method’
  229 |   rb_define_method(klass,name.c_str(),RUBY_METHOD_FUNC(get), missing ? -1 : 0);
      |   ^~~~~~~~~~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:61: error: ‘missing’ is not a constant expression
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                             ^
rwx.cpp:231:3: note: in expansion of macro ‘rb_define_method’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |   ^~~~~~~~~~~~~~~~
rwx.cpp:231:78: note: in template argument for type ‘int’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |                                                                      ~~~~~~~~^~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:60: note: in definition of macro ‘rb_define_method’
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                            ^
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: warning: expression list treated as compound expression in initializer [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
rwx.cpp:231:3: note: in expansion of macro ‘rb_define_method’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |   ^~~~~~~~~~~~~~~~
/usr/include/ruby-2.7.0/ruby/ruby.h:2777:78: warning: invalid conversion from ‘VALUE (*)(...)’ {aka ‘long unsigned int (*)(...)’} to ‘int’ [-fpermissive]
 2777 | #define rb_define_method(m, n, f, a) rb_define_method_tmpl<a>::define(m, n, f)
      |                                                                              ^
      |                                                                              |
      |                                                                              VALUE (*)(...) {aka long unsigned int (*)(...)}
rwx.cpp:231:3: note: in expansion of macro ‘rb_define_method’
  231 |   rb_define_method(klass,(name + "=").c_str(),RUBY_METHOD_FUNC(set), missing ? -1 : 1);
      |   ^~~~~~~~~~~~~~~~
make: *** [Makefile:237 : rwx.o] Erreur 1
</pre>

But after my change to ext/rwx.cpp, it now compiles without -fpermissive :-). Sorry I didn't check that again. I will amend my pull update accordingly!

--- a/ext/extconf.rb
+++ b/ext/extconf.rb
@@ -69,7 +69,8 @@ if(wxversion = pkg_config('wx', 'version'))
 		$CXXFLAGS << " " << pkg[0] if pkg && !$CXXFLAGS[pkg[0]]
 	}
 	
-	all = " -fvisibility-inlines-hidden -fpermissive"
+	all = " -fvisibility-inlines-hidden"
+
 	$CFLAGS << all << " -x c++ -g "
 	$CXXFLAGS << all << " -g "
 	$CPPFLAGS << all << " -g "

